Exclude flat checks from adjusting `inline-dc`

In3luki opened the pull request:
Required to exclude some check types (e.g. flat check) from being adjusted by rule elements.

In3luki committed:
Add check type to inline DC roll options

In3luki committed:
Exclude flat checks from `inline-dc` adjustments

In3luki commented:
Now excludes all flat checks from being modified by the `inline-dc` statistic. I've kept the type because that might still be useful but it can be removed if necessary.

CarlosFdez commented:
We might still want a statistic for it, but with a more specific selector. We should make sure nothing is relying on it though.

I'd prefer to avoid adding the extra option since they're hard to take-backsie, but I don't feel too strongly about that.

In3luki commented:
Updated to give inline flat checks the `flat-check` domain exclusively. Should that be `inline-flat-check`?
Also I found no rule elements that modify the flat check dc value in the system at the moment.
Also removed the `type` roll option. I think it's not really required anymore at this point.

In3luki committed:
Rename to `flat-check-dc`
